### ISSUE TITLE
Fix 'test_feeds/test_invalid_values' tests for Debian, Rehat and Canonical in 4.2

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#1548]
### Issue information

| Related issue |
|--|
| Closes https://github.com/wazuh/wazuh-qa/issues/1531|

When it comes to starting with the test fix, there is some information that can help you:
```
Module: Vulnerability Detector
Support Team:  "binary beasts"
Target: Manager   
OS: Linux
```
I add some information obtained while researched the logs that failed.

## Case 1
|  Type | Description  |
|--|--|
|Test Path | test_feeds/debian/test_invalid_values_debian_feed
| Consistent | no
| Test Execution | 1/3 executions failed
| Cases Fails | 54
| Summary | On `modify_feed` when it called `read_xml_file` method is called. The problem seems to be a Parser error when reading the file.
| Error message |  "ERROR at setup of test_invalid_values_debian_feed" 

## Case 2

|  Type | Description  |
|--|--|
|Test Path | test_feeds/redhat/test_invalid_values_redhat_feed
| Consistent | no
| Test Execution | 2/3 executions failed
| Cases Fails | 140
| Summary | On `modify_feed` when it called `read_xml_file` method is called. 
| Error message | "ERROR at setup of test_invalid_redhat_feed"

## Case 3

|  Type | Description  |
|--|--|
|Test Path | test_feeds/canonical/test_invalid_values_canonical_feed
| Consistent | yes
| Test Execution | 3/3 executions failed
| Cases Fails | 166
| Summary |  - On `modify_feed` method when it called to `read_xml_file`. The problem seems to be a Parser error when reading the file.
|| - On `check_failure_when_importing_feed` method. Research if it's an error when import the file or if it's a Parser error in log event.|
| Error message |  "Could not find the log event: ERROR: \\(\\d+\\): CVE database could not be updated."

In order to finish this issue the following tasks should be fulfilled:
- [ ] Research of fails.
- [ ] Apply Fix.
- [ ] Full Green/ Full Yellow  in `test_feeds/debian/test_invalid_values_debian_feed` for 3 times. 
- [ ] Full Green/ Full Yellow  in `test_feeds/redhat/test_invalid_values_redhat_feed` for 3 times.  
- [ ] Full Green/ Full Yellow  in `test_feeds/canonical/test_invalid_values_canonical_feed` for 3 times. 
- [ ] Documentation of any important change done for these tests or the used tools.

**Comment by mdengra** [comment]
### 2021-07-19

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the modules `sca`, `syscollector` and `rootcheck` modules disabled in the `ossec.conf`: 

### Case 1
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_debian_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6841580/test_invalid_values_debian_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_values_debian_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6841581/test_invalid_values_debian_feed_local_dm_r2.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_values_debian_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6841654/test_invalid_values_debian_feed_local_dm_r3.log)| 2021-07-19 | Miguel | :yellow_circle: 


### Case 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_redhat_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6841584/test_invalid_values_redhat_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_values_redhat_feed_local_dm_r2.log](https://github.com/wazuh/wazuh-qa/files/6841858/test_invalid_values_redhat_feed_local_dm_r2.log)| 2021-07-19 | Miguel | :yellow_circle: 
|[test_invalid_values_redhat_feed_local_dm_r3.log](https://github.com/wazuh/wazuh-qa/files/6841862/test_invalid_values_redhat_feed_local_dm_r3.log)| 2021-07-19 | Miguel | :yellow_circle: 


### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_canonical_feed_local_dm_r1.log](https://github.com/wazuh/wazuh-qa/files/6841874/test_invalid_values_canonical_feed_local_dm_r1.log)| 2021-07-19 | Miguel | :yellow_circle: 
|| 2021-07-19 | Miguel | :yellow_circle: 
|| 2021-07-19 | Miguel | :yellow_circle:

**Comment by mdengra** [comment]
### 2021-07-22

Used Wazuh-QA branch: `1531-full-yellow-vuln-det`
Test results with the default settings in the `ossec.conf`: 

### Case 1
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_debian_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6861907/test_invalid_values_debian_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_values_debian_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6861908/test_invalid_values_debian_feed_local_r2.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_values_debian_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6861909/test_invalid_values_debian_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle: 


### Case 2
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_redhat_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6862152/test_invalid_values_redhat_feed_local_r1.log)| 2021-07-22 | Miguel | :red_circle: 
|[test_invalid_values_redhat_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6862154/test_invalid_values_redhat_feed_local_r2.log)| 2021-07-22 | Miguel | :red_circle: 
|[test_invalid_values_redhat_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6862155/test_invalid_values_redhat_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle: 

Note that the same use case (first case) fails in the two first runs:
```
FAILED test_vulnerability_detector/test_feeds/redhat/test_invalid_values_redhat_feed.py::test_invalid_redhat_feed[REDHAT_configuration-tag: definitions, value: None]
```

### Case 3
| Test Executions | Date  | By  | Status | 
|--|--|--|--|
|[test_invalid_values_canonical_feed_local_r1.log](https://github.com/wazuh/wazuh-qa/files/6862294/test_invalid_values_canonical_feed_local_r1.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_values_canonical_feed_local_r2.log](https://github.com/wazuh/wazuh-qa/files/6862297/test_invalid_values_canonical_feed_local_r2.log)| 2021-07-22 | Miguel | :yellow_circle: 
|[test_invalid_values_canonical_feed_local_r3.log](https://github.com/wazuh/wazuh-qa/files/6862299/test_invalid_values_canonical_feed_local_r3.log)| 2021-07-22 | Miguel | :yellow_circle:

**Comment by mdengra** [comment]
#### Same scenario as in the issue: https://github.com/wazuh/wazuh-qa/issues/1547
The problem is related to the issue: #1602, in which FileMonitor does not detect a message written to the log file under certain scenarios.

In the test_invalid_syntax_redhat_feed.py test, even disabling all possible modules before launching the test does not prevent it from falling in the first use case.

The cause of the failure is the large number of messages generated by enabling the internal option wazuh_modules.debug=2, which is required by multiple vulnerability detector tests. When this option is enabled and the manager is restarted, the log file takes up 1MB, this added to the messages generated by the test itself makes FileMonitor unable to process them, causing the error.

**Comment by damarisg** [comment]
Closed by #1631 

### Test Results after to merge on branch `1531-full-yellow-vuln-det`

| Test Executions | Date  | Tests | By  | Status | 
|--|--|--|--|--|
|[YellowInvalidValuesDebianFeed.log](https://github.com/wazuh/wazuh-qa/files/6870553/YellowInvalidValuesDebianFeed.log)| 2021-07-23 | test_feeds/debian/test_invalid_values_debian_feed | Seyla | 🟡 
|[YellowInvalidValuesDebianFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870554/YellowInvalidValuesDebianFeed2.log)| 2021-07-23 | test_feeds/debian/test_invalid_values_debian_feed | Seyla | 🟡 
|[YellowInvalidValuesDebianFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870560/YellowInvalidValuesDebianFeed3.log)| 2021-07-23 | test_feeds/debian/test_invalid_values_debian_feed | Seyla | 🟡 
|[YellowInvalidValuesRedhatFeed.log](https://github.com/wazuh/wazuh-qa/files/6870652/YellowInvalidValuesRedhatFeed.log)| 2021-07-22 | test_feeds/redhat/test_invalid_values_redhat_feed | Seyla | 🟡 
|[YellowInvalidValuesRedhatFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870653/YellowInvalidValuesRedhatFeed2.log)| 2021-07-22 | test_feeds/redhat/test_invalid_values_redhat_feed | Seyla | 🟡 
|[YellowInvalidValuesRedhatFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870654/YellowInvalidValuesRedhatFeed3.log)| 2021-07-22 | test_feeds/redhat/test_invalid_values_redhat_feed | Seyla | 🟡 
|[YellowInvalidValuesCanonicalFeed.log](https://github.com/wazuh/wazuh-qa/files/6870874/YellowInvalidValuesCanonicalFeed.log)| 2021-07-22 | test_feeds/canonical/test_invalid_values_canonical_feed | Seyla | 🟡 
|[YellowInvalidValuesCanonicalFeed2.log](https://github.com/wazuh/wazuh-qa/files/6870875/YellowInvalidValuesCanonicalFeed2.log)| 2021-07-22 | test_feeds/canonical/test_invalid_values_canonical_feed | Seyla | 🟡 
|[YellowInvalidValuesCanonicalFeed3.log](https://github.com/wazuh/wazuh-qa/files/6870876/YellowInvalidValuesCanonicalFeed3.log)| 2021-07-22 | test_feeds/canonical/test_invalid_values_canonical_feed | Seyla | 🟡
